python/bind_operators.h

```diff
@@ -123,6 +127,18 @@ template <class RHSSetup> struct OpBinder {
           return a;
         },
         py::is_operator(), py::call_guard<py::gil_scoped_release>());
+    if constexpr (!(std::is_same_v<T, Dataset> ||
```

Comment:
Minor point: Maybe we should have a local templated `using DatasetOrView` to avoid this repeated || in the bindings?

Reply:
Not worth it. This will be removed once we extend the generators to make the functions for `Dataset` in addition to `DataArray`.